kubernetes/deployment.template.yaml

```diff
@@ -0,0 +1,216 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ${SERVICE_NAME}
+  namespace: ${NAMESPACE}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ${SERVICE_NAME}
+  template:
+    metadata:
+      labels:
+        app: ${SERVICE_NAME}
+    spec:
+      initContainers:
+        - name: init-wait-db
+          image: busybox
+          command: ['sh', '-c', 'for i in $(seq 1 300); do nc -zvw1 shared-database 5432 && exit 0 || sleep 3; done; exit 1']
+      containers:
+        - image: ${IMAGE_NAME}
+          name: ${SERVICE_NAME}
+          envFrom:
+            - configMapRef:
+                name: ${SERVICE_NAME}-config
+          env:
+#            Import variables from vault
+            - name: S2S_KEY
+              value: '${S2S_TOKEN}'
+#            Import from main.tf
+#            This is to use the AAT database
+            - name: POSTGRES_HOST
+              value: "${POSTGRES_HOST}"
+            - name: POSTGRES_DATABASE
+              value: "${POSTGRES_DATABASE}"
+            - name: POSTGRES_USER
+              value: "${POSTGRES_USER}"
+            - name: POSTGRES_PASSWORD
+              value: "${POSTGRES_PASSWORD}"
+
+#            db
+            - name: SPRING_DATASOURCE_URL
+              value: 'jdbc:postgresql://shared-database:5432/emannotationapp?ssl=true'
+            - name: SPRING_DATASOURCE_USERNAME
+              value: 'emannotationapp'
+            - name: SPRING_DATASOURCE_PASSWORD
+              value: 'emannotationapp'
+            - name: ENABLE_DB_MIGRATE
+              value: 'true' #TODO: Alec set it as true, but main.tf has value = false. What should I set it to?
```

Comment:
true is correct

Reply:
so how come main.tf has false? Should I change that to true?

Reply:
no, main.tf has a zero down time deployment approach with staging and production slots.
and a special stage for migrating the database, this doesn't have any of that so you need the app to migrate it on startup